src/m365/graph/commands/changelog/changelog-list.ts

```diff
@@ -0,0 +1,198 @@
+      headers: {
+        'accept': 'text/xml',
+        'x-anonymous': 'true'
+      }
+    };
+
+    request
+      .get(requestOptions)
+      .then((output: any) => {
+        const parser = new DOMParser();
+        const xmlDoc = parser.parseFromString(output.toString(), "text/xml");
+
+        const changelog = this.filterThroughOptions(args.options, this.mapChangelog(xmlDoc));
+        
+        logger.log(changelog.items);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private filterThroughOptions(options: Options, changelog: Changelog): Changelog {
+    let items: ChangelogItem[] = changelog.items;
+
+    if (options.services) {
+      const allowedServices: string[] = this.allowedServices
+        .filter(allowedService => options.services!.toLocaleLowerCase().split(',').includes(allowedService.toLocaleLowerCase()));
+
+      items = changelog.items.filter(item => allowedServices.includes(item.title));
+    }
+
+    if (options.versions) {
+      const allowedVersions: string[] = this.allowedVersions
+        .filter(allowedVersion => options.versions!.toLocaleLowerCase().split(',').includes(allowedVersion.toLocaleLowerCase()));
+
+      items = items.filter(item => allowedVersions.includes(item.category));
+    }
+
+    if (options.startDate) {
+      const startDate: Date = new Date(options.startDate);
+
+      items = items.filter(item => item.pubDate >= startDate);
+    }
+
+    if (options.endDate) {
+      const endDate: Date = new Date(options.endDate);
+
+      items = items.filter(item => item.pubDate <= endDate);
+    }
+
+    // Make sure everything is unique based on the item guid
+    items = [...new Map(items.map((item) => [item.guid, item])).values()];
+
+    // Order items by date desc
+    changelog.items = items.sort((itemA, itemB) => Number(itemB.pubDate) - Number(itemA.pubDate));
+
+    return changelog;
+  }
+
+  private mapChangelog(xmlDoc: any): Changelog {
+    const channel = xmlDoc.getElementsByTagName('channel').item(0);
+    
+    const changelog: Changelog = {
+      title: channel.getElementsByTagName('title').item(0).textContent,
+      description: channel.getElementsByTagName('description').item(0).textContent,
+      url: channel.getElementsByTagName('link').item(0).textContent,
+      items: []
+    } as Changelog;
+
+    Array.from(xmlDoc.getElementsByTagName('item')).forEach((item: any) => {
+      changelog.items.push({
+        guid: item.getElementsByTagName('guid').item(0).textContent,
+        category: item.getElementsByTagName('category').item(1).textContent,
+        title: item.getElementsByTagName('title').item(0).textContent,
+        description: item.getElementsByTagName('description').item(0).textContent,
+        pubDate: new Date(item.getElementsByTagName('pubDate').item(0).textContent)
+      } as ChangelogItem);
+    });
+
+    return changelog;
+  }
```

Comment:
if we done our own mapper from xml to json do we need the `xml2js` package 🤔? I mean where it is used ?

Reply:
Well, it isn't used anywhere. 😅 I forgot to remove it.